### PR TITLE
Added joining lobbies via short codes

### LINKED ISSUE #8: Join room with a code
Add way to join rooms with a short code

### PR DESCRIPTION
**Pull request by yohanc3**
**TLDR: Lobbies can be joined via 4 digit codes.**

Now each lobby has its own unique 4 digit code. Users can join matches by passing a 4 digit code, if it's not found a toast will pop up, same if the code is invalid

**UI changes:** 

1. Added Join room section Component

join room component

3. Added in-game lobby-code button

ingame code added

To consider: 

- I was originally planning on placing the Join room component under the Create Room form but you could barely see it on the screen. I would consider placing it at the bottom since if users were to actually use it, it would mean there are at least +12 ongoing games, which would force them to scroll down to look for the one their looking for.

### CODE REVIEW

**Comment by heathhenley** [comment]
Sweet! FYI @webdevcody I pulled this in and locally and it's working great.  The only nitpicky thing I would recommend is just running prettier on the changed files to match the format

**Review by heathhenley — APPROVED** [review]
@webdevcody was crazy enough to give me push access, are you set up to be able to format with prettier before merge? No worries if not I could push it (but I can't look at it until later today / tomorrow)

**Comment by yohanc3** [comment]
@heathhenley now it should be ready to merge, just ran prettier on the files I previously modified. Will merge upon approval, thanks!

**Review by heathhenley — APPROVED** [review]
